pkg/apis/acme/v1/types_issuer.go

```diff
@@ -482,19 +482,37 @@ type ACMEIssuerDNS01ProviderAzureDNS struct {
 	// +optional
 	ClientSecret *cmmeta.SecretKeySelector `json:"clientSecretSecretRef,omitempty"`
 
+	// ID of the Azure subscription
 	SubscriptionID string `json:"subscriptionID"`
 
 	// when specifying ClientID and ClientSecret then this field is also needed
 	// +optional
 	TenantID string `json:"tenantID,omitempty"`
 
+	// resource group the DNS zone is located in
 	ResourceGroupName string `json:"resourceGroupName"`
 
+	// name of the DNS zone that should be used
 	// +optional
 	HostedZoneName string `json:"hostedZoneName,omitempty"`
 
+	// name of the Azure environment (default AzurePublicCloud)
 	// +optional
 	Environment AzureDNSEnvironment `json:"environment,omitempty"`
+
+	// managed identity configuration, can not be used at the same time as clientID, clientSecretSecretRef or tenantID
+	// +optional
+	ManagedIdentity *AzureManagedIdentity `json:"managedIdentity,omitempty"`
```

Comment:
Yeah, I prefer it this way. Thanks for making the change.

I have no experience with Azure....are there any other managed identity fields which we are likely to need  in the future. Will they fit intuitively into this struct?

Reply:
I'm not sure, I can't think of any. The only reason we need these is that clusters with multiple managed identities need a way to select which one should be used.